WordPress/src/main/java/org/wordpress/android/ui/reader/reblog/ReblogState.kt

```diff
@@ -0,0 +1,38 @@
+package org.wordpress.android.ui.reader.reblog
+
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.models.ReaderPost
+
+/**
+ * Represents the Reblog View State
+ *
+ * @property site the site to reblog to
+ * @property post the post to be reblogged
+ */
+sealed class ReblogState(val site: SiteModel? = null, val post: ReaderPost? = null)
+
+/**
+ * Represents the Site Picker View State
+ *
+ * @property site the preselected site
+ * @property post the post to be reblogged (saved for later)
+ */
+class SitePicker(site: SiteModel, post: ReaderPost) : ReblogState(site, post)
+
+/**
+ * Represents the Post Editor View State
+ *
+ * @property site the site to reblog to
+ * @property post the post to be reblogged
+ */
+class PostEditor(site: SiteModel, post: ReaderPost) : ReblogState(site, post)
+
+/**
+ * Represents the No Site View State
+ */
+object NoSite : ReblogState()
+
+/**
+ * Represents the Error State
+ */
+object ReblogError : ReblogState()
```

Comment:
> Having this last consideration, since NoSite and ReblogError are not keeping state, what do you think about moving site and post only to SitePicker and PostEditor? 

You are right, The NoSite/ReblogError cases do not keep any state. I've found it useful in the past to have some common properties in the top level sealed class (even if they are not used in some cases) to be able to handle states in a uniform way or to avoid ugly casting especially in java. Type safety is an undesired tradeoff of course.

> side note: in general I would also consider to create a null object state here like Unknown or similar, but in this case we are going to use it in a live variable in kotlin code so maybe not much value added

My intention was to use the ErrorState as Unknown. Probably my naming is bad. I'm not sure if both can fit in our case.

I made fixes based on your feedback with 490ae8aab850f4f0e2f9c1e6c38b54d8a80eac07

Reply:
> ...avoid ugly casting especially in java

Agree; just to share maybe could make sense in cases like this to have a type enum in the base sealed class and manage the two with state classes as a data class (discriminated by the type that should work in Java also)